Mathlib/GroupTheory/GroupAction/Amenable.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2023 Matthias Uschold. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matthias Uschold
+-/
+import Mathlib.GroupTheory.GroupAction.Defs
+import Mathlib.Data.Real.ENNReal
+
+
+/-!
+# Amenable Monoid Actions
+
+In this file, we define amenability of a monoid action.
+A monoid action is amenable if it admits a invariant mean.
+A mean is like a probability measure, demanding
+only finite additivity instead of σ-additivity.
+
+## Main Definitions
+
+- `Mean`: defines means as finitely additive probability measures
+- `InvariantMean`: defines when a mean is invariant under a monoid action
+- `amenable`: A monoid action is amenable if there exists an invariant mean
+
+## Implementation Notes
+
+`Mean` was not implemented using MeasureTheory.ProbabilityMeasure
+because meausures are σ-additive (i.e. countably additive). In this
+setting, this would be a too strong assumption, we only want to demand
+finite additivity. Typically, the resulting measures will not be
+σ-additive.
+
+## References
+
+See [bartholdi2017amenability] for definitions
+and more information on amenable actions.
+
+-/
+
+universe u v
+variable (M : Type u) (α : Type v) [Monoid M] [MulAction M α]
+
+
+/--A mean is a function from the power set of α to ENNReal that
+- assigns the value 1 to the full set α, and
+- is finitely additive under disjoint unions -/
+structure Mean where
+  /-- function giving the measure of subset-/
+  μ : Set α  → ENNReal
+  /-- μ should be normalised  -/
+  norm : μ Set.univ = 1
+  /-- μ has to be finitely additive -/
+  fin_add : ∀ (X Y : Set α), Disjoint X Y
```

Comment:
It feels wrong to require this on all sets, and not just on sets in some sigma-algebra. (Of course, for discrete groups there is no issue). See for instance appendix G of the book by Bekka, de la Harpe and Valette for the "right" general point of view. (I know that Bartholdi is more into discrete groups, and so am I, but for many applications the greater generality is necessary).

Reply:
Thanks for pointing this out! I'll work on this ...

Reply:
I adapted the situation to the sigma-algebra setting now. Could you take a look at it?